From 86c4d9d0fd044a17f5be442ad2875c9d8681a261 Mon Sep 17 00:00:00 2001
From: "https://lore.kernel.org/all/20230518184541.2627-7-jszhang@kernel.org/"
 <https://lore.kernel.org/all/20230518184541.2627-7-jszhang@kernel.org/>
Date: Thu, 21 Sep 2023 16:47:50 +0800
Subject: [PATCH 2/4] th1520: add cpu reset node

+		cpurst: cpurst {
+			compatible = "thead,reset-th1520"; -> "thead,reset-sample";
+			entry-reg = <0xff 0xff019050>;
+			entry-cnt = <4>;
+			control-reg = <0xff 0xff015004>;
+			control-val = <0x1c>;
+			csr-copy = <0x7f3 0x7c0 0x7c1 0x7c2 0x7c3 0x7c5 0x7cc>;
+		};
+

Upstream-Status: Pending
Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
Signed-off-by: Kanak Shilledar <kanakshilledar111@protonmail.com>
---
 arch/riscv/boot/dts/thead/th1520.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/riscv/boot/dts/thead/th1520.dtsi b/arch/riscv/boot/dts/thead/th1520.dtsi
index 527336417765..4f8d5bdf396c 100644
--- a/arch/riscv/boot/dts/thead/th1520.dtsi
+++ b/arch/riscv/boot/dts/thead/th1520.dtsi
@@ -237,6 +237,15 @@ soc {
 		dma-noncoherent;
 		ranges;
 
+		cpurst: cpurst {
+			compatible = "thead,reset-sample";
+			entry-reg = <0xff 0xff019050>;
+			entry-cnt = <4>;
+			control-reg = <0xff 0xff015004>;
+			control-val = <0x1c>;
+			csr-copy = <0x7f3 0x7c0 0x7c1 0x7c2 0x7c3 0x7c5 0x7cc>;
+		};
+
 		plic: interrupt-controller@ffd8000000 {
 			compatible = "thead,th1520-plic", "thead,c900-plic";
 			reg = <0xff 0xd8000000 0x0 0x01000000>;
-- 
2.43.0

